R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -196,6 +196,12 @@ test_that("create DataFrame from RDD", {
   expect_equal(dtypes(df), list(c("name", "string"), c("age", "int"), c("height", "float")))
   expect_equal(as.list(collect(where(df, df$name == "John"))),
                list(name = "John", age = 19L, height = 176.5))
+  expect_equal(getNumPartitions(toRDD(df)), 1)
```

Comment:
Hmm, good point, the behavior is a bit strange and I haven't thought of a concise way to document this.
https://github.com/apache/spark/blob/master/R/pkg/R/context.R#L131

Basically, it's the largest of numSlices or ceiling of data size divided by `spark.r.maxAllocationLimit` - *but* limited by length of the data (but this length is wrong if the data is a data.frame - since that length becomes the number of columns).

Is this an unintentional behavior (ie. limited always by the number of columns even when the data size is larger then the `spark.r.maxAllocationLimit`)? I can't tell...